build_tools/third_party/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set(TARGET_DIR ${CMAKE_CURRENT_BINARY_DIR})
+set(DOWNLOAD_SCRIPT_URL "https://raw.githubusercontent.com/NVIDIA/build-system-archive-import-examples/44dfb51fad75a8a2f1044a4fe221aba70571b86f/parse_redist.py")
+set(DOWNLOAD_DIR ${TARGET_DIR}/download)
+set(DOWNLOAD_SCRIPT_PATH ${DOWNLOAD_DIR}/parse_redist.py)
+
+# Parameters to the download script.
+# Look for an appropriate redistrib_*.json here to verify:
+#   https://developer.download.nvidia.com/compute/cuda/redist/
+set(VERSION "11.6.2")
+set(PRODUCT "cuda")
+if(UNIX)
+  set(OS "linux")
+elseif(WIN32)
+  set(OS "windows")
+else()
+  message(SEND_ERROR "Unsupported OS environment. Must be Windows or Linux.")
+  return()
+endif()
+set(ARCH "x86_64")
+
+# Components that we need to fetch.
+unset(COMPONENTS)
+list(APPEND COMPONENTS "cuda_nvcc")
+list(APPEND COMPONENTS "cuda_cudart")
+
+# Paths within the arch specific installation that we want to retain.
+unset(RETAIN_PATHS)
+list(APPEND RETAIN_PATHS "LICENSE")
+list(APPEND RETAIN_PATHS "nvvm/libdevice/libdevice.10.bc")
+list(APPEND RETAIN_PATHS "include/cuda.h")
+
+message(NOTICE "Extracting to ${TARGET_DIR}")
```

Comment:
nit: `STATUS`, we don't have other uses of `NOTICE`. You might also want to add more context to these messages (like mention "CUDA" specifically)

https://cmake.org/cmake/help/latest/command/message.html

```
(none) or NOTICE
Important message printed to stderr to attract user's attention.

STATUS
The main interesting messages that project users might be interested in. Ideally these should be concise, no more than a single line, but still informative.
```